### DISCUSSION TITLE
Parse Json Response

### DISCUSSION THREAD

**Discussion by nirvine** [Waboodoo/HTTP-Shortcuts#243] [Q&A] [answered]
I would like to get the temperature from a Sensibo smart air conditioning device, but cannot seem to work out the structure of the json parse statement. I have used the following in the On Success section: 

```
const temperature = JSON.parse(response.body).result.temperature;
setVariable('temperature',temperature);
showDialog('Indoor Temp: '+temperature+' °C', 'Temperature');
```
The json data is:
{"status": "success", "result": [{"time": {"time": "2021-10-09T08:41:20.666359Z", "secondsAgo": 71}, "temperature": 23.3, "humidity": 60.7, "feelsLike": 23.3, "rssi": -58, "motion": false, "roomIsOccupied": true}]}

Output is "Indoor Temp: undefined C"

What have I done wrong?
And the next step would be to include the humidity in the output, but temperature is more important to me.

"

**Comment by Waboodoo** [comment, marked as answer]
Change the first line to the following:

```js
const temperature = JSON.parse(response.body).result[0].temperature;
````

"result" is an array with one element, so you need to select the 0-th element from it before you try to access `temperature`.
The same should work for accessing the humidity:

```js
const result = JSON.parse(response.body).result[0];
const temperature = result.temperature;
const humidity = result.humidity;
```

Also, you probably don't need the second line of your code (where you call `setVariable`), unless you actual have a global variable called "temperature" set up in the app which you are using somewhere else in the same shortcut (e.g. in a header) or in some other place. If you only need the temperature value right inside the JS code then a JS variable is enough and you don't need to go with `getVariable()` and `setVariable()`.

**Reply by nirvine** [reply]
Many thanks - it is working now. I could not find a guide on the format of the parse statement.

**Reply by Waboodoo** [reply]
You can find more information about `JSON.parse()` here for example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse
